### PR TITLE
Fixed auto-login does not work with enabled page cache.

### PR DESCRIPTION
**Pull request by juliquiron**
Fixed autologin not performing the login.

### CODE REVIEW

File: src/EventSubscriber/SsofactAutoLoginSubscriber.php [hunk 1]
```diff
@@ -55,11 +55,11 @@ public function __construct(
...
     // Run after:
     // - 300: AuthenticationSubscriber (to get current_user)
-    // - 30: MaintenanceModeSubscriber (only local accounts may login during
-    //   maintenance)
     // Run before:
+    // - 30: MaintenanceModeSubscriber (admins should be able to log in during
+    //   maintenance)
     // - 27: DynamicPageCacheSubscriber
-    $events[KernelEvents::REQUEST][] = ['onKernelRequestAuthenticate', 29];
+    $events[KernelEvents::REQUEST][] = ['onKernelRequestAuthenticate', 290];
```
Comment (sun) [inline src/EventSubscriber/SsofactAutoLoginSubscriber.php]: This change might not be necessary anymore

**Review by sun — APPROVED** [review]

**Comment by sun** [comment]
This fix no longer works for some reason…

1. One simple problem was that the integration branch of the ssofact module did not have the latest master branch merged-in, so the change was actually missing.

1. But even with the fix and after clearing all caches, the page response of https://dev.stimme.de/user appears to be cached and always redirects to /user/login even when the user has the SSO login cookie set (which should cause a redirect into the authentication process instead). — I'm not able to see why that is.

/cc @rodrigoaguilera

**Comment by sun** [comment]
When registering, the redirect response on /user is coming from cache:
```
POST https://stage-login.stimme.de/index.php?next=eJzLKCkpKLbS109JLdMrLsnMzU3VS0nVLy1OLQIAggAJrQ%2C%2C&redirect_url=https%3A%2F%2Fdev.stimme.de%2Fshop%2Fuser%2Faccount&pageid=52
<>
location: https://dev.stimme.de/user
set-cookie: RF_OAUTH_SERVER=...%2C%2C; path=/; domain=.stimme.de; secure; HttpOnly
status: 303


GET https://dev.stimme.de/user
location: https://dev.stimme.de/user/login
status: 302
x-drupal-cache: HIT
```

**Comment by rodrigoaguilera** [comment]
I think I am missing more context about this problem before trying to analyze it.

Maybe some steps to reproduce.
Should I have the sso configured? What test users should I use? What is normal login flow?

**Comment by sun** [comment]
Yes, you would need to have the SSO set up locally to reproduce it locally, but note that the SSO login cookie is only set on *.stimme.de domains, so you will not be logged in on your local sites.

It's probably easiest to debug this directly on dev.stimme.de?  Can you look into this today?

**Comment by sun** [comment]
@rodrigoaguilera found the cause and proper fix in https://github.com/netzstrategen/drupal-ssofact/pull/17 👍

**Comment by rodrigoaguilera** [comment]
@juliquiron was guiding me most of the time. He deserves the credit 🙇

**Pull request merged**